src/robotFactory.js

```diff
@@ -1,17 +1,67 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype = {
+  goForward(step = 1) {
+    this.coords.y += step;
+  },
+  goBack(step = 1) {
+    this.coords.y -= step;
+  },
+  goRight(step = 1) {
+    this.coords.x += step;
+  },
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  },
+  getInfo() {
+    return `Robot: ${this.name}, `
+          + `Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  },
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, ...arguments);
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype = {
+  goUp(step = 1) {
+    this.coords.z += step;
+  },
+  goDown(step = 1) {
+    this.coords.z -= step;
+  },
+};
+
+Object.setPrototypeOf(FlyingRobot.prototype, BaseRobot.prototype);
+
+function DeliveryDrone(name, weight, coords, chipVersion,
+  maxLoadWeight, currentLoad) {
```

Comment:
```suggestion
function DeliveryDrone(
  name,
  weight,
  coords,
  chipVersion,
  maxLoadWeight,
  currentLoad) {
```
В стовпчик дійсно легше читається

Reply:
скобку лучше тоже переносить